src/halmos/sevm.py

```diff
@@ -644,8 +644,10 @@ class Concretization:
     In the future, it may be used for other purposes, such as concrete hash reasoning.
     """
 
-    substitution: dict = field(default_factory=dict)  # symbol -> constant
-    candidates: dict = field(default_factory=dict)  # symbol -> set of constants
+    # symbol -> constant
+    substitution: dict[BitVecRef, BitVecRef] = field(default_factory=dict)
```

Comment:
is this actually `substitution: dict[BitVecRef, int]`?

Reply:
are there any constraints on the bit size of keys in `substitution`?

Reply:
no, the values are bv_value. no constraints on the bitsize.